build.gradle

```
@@ -255,51 +255,41 @@ if (gradle.startParameter.taskNames == ["travisCiBuild"]) {
 
 if (gradle.startParameter.taskNames == ["ghActionsBuild"]) {
   gradle.startParameter.taskNames = ["build", "codeCoverageReport"]
-  subprojects {
-    tasks.withType(Test) {
-      maxParallelForks = 2
-    }
-  }
+}
 
-  if (System.getenv("GITHUB_BASE_REF") == null                                // GITHUB_BASE_REF is only set for Pull requests
+if (gradle.startParameter.taskNames == ["ghActionsPublish"] || gradle.startParameter.taskNames == ["ghActionsDocs"] ) {
+  gradle.startParameter.taskNames = []
+  if (!(System.getenv("GITHUB_EVENT_NAME") == "push"
           && System.getenv("GITHUB_REF") == "refs/heads/master"
-          && System.getenv("GITHUB_REPOSITORY") == 'spockframework/spock') {
-    if (javaVersion == javaVersions.min()) {
-      gradle.startParameter.taskNames += ["publish"]
-      if (!snapshotVersion) {
-        gradle.startParameter.taskNames += ["tagRelease"]
-      }
+          && System.getenv("GITHUB_REPOSITORY") == 'spockframework/spock')) {
+    throw new IllegalArgumentException("""ghActionsPublish can only be run on branch master in repo spockframework/spock on push
```

+1 for the exception itself.

```suggestion
    throw new IllegalArgumentException("""ghActionsPublish can only be run on push to branch master in repo spockframework/spock
```
?